File: mvp_rxjava_retrofit/src/main/java/com/example/mvp_rxjava_retrofit/base/BaseApp.java
package com.example.mvp_rxjava_retrofit.base;

import android.app.Application;

public class BaseApp extends Application {
    private static BaseApp instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public static BaseApp getInstance() {
        return instance;
    }
}

File: mvi/src/main/java/com/example/mvi/common/app/BaseApp.kt
package com.example.mvi.common.app

import android.app.Application
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class BaseApp : Application() {

    companion object {
        private lateinit var instance: BaseApp

        fun getInstance() = instance
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }
}